packages/graphql/src/schema/make-augmented-schema.ts

```diff
@@ -121,6 +124,83 @@ function makeAugmentedSchema(
         },
     });
 
+    const composeInt = {
+        type: "Int!",
+        resolve: numericalResolver,
+        args: {},
+    };
+
+    const composeFloat = {
+        type: "Float!",
+        resolve: numericalResolver,
+        args: {},
+    };
+
+    const composeId = {
+        type: "ID!",
+        resolve: idResolver,
+        args: {},
+    };
+
+    // Foreach i if i[1] is ? then we will assume it takes on type { min, max }
+    const aggregationSelectionTypeMatrix: [string, any?][] = [
+        [
+            "ID",
+            {
+                shortest: composeId,
+                longest: composeId,
+            },
+        ],
+        [
+            "String",
+            {
+                shortest: "String!",
+                longest: "String!",
+            },
+        ],
+        [
+            "Float",
+            {
+                max: composeFloat,
+                min: composeFloat,
+                average: composeFloat,
+            },
+        ],
+        [
+            "Int",
+            {
+                max: composeInt,
+                min: composeInt,
+                average: composeFloat,
+            },
+        ],
+        [
+            "BigInt",
+            {
+                max: "BigInt!",
+                min: "BigInt!",
+                average: "BigInt!",
+            },
+        ],
+        ["DateTime"],
+        ["LocalDateTime"],
+        ["LocalTime"],
+        ["Time"],
+        ["Duration"],
+    ];
+
+    const aggregationSelectionTypeNames = aggregationSelectionTypeMatrix.map(([name]) => name);
+
+    const aggregationSelectionTypes = aggregationSelectionTypeMatrix.reduce((res, [name, fields]) => {
```

Comment:
You _should_ be able to get rid of the hacky `as unknown) as ObjectTypeComposer<unknown, unknown>;` from below if you do:

```suggestion
    const aggregationSelectionTypes = aggregationSelectionTypeMatrix.reduce<Record<string, ObjectTypeComposer<unknown, unknown>>>((res, [name, fields]) => {
```

Cleaner to specify your types up front.